qcelemental/models/molecule.py

```diff
@@ -738,6 +739,12 @@ def from_data(cls,
         else:
             raise KeyError("Dtype not understood '{}'.".format(dtype))
 
+        input_dict.update(kwargs)
+        if "molecular_charge" in kwargs and "fragment_charges" not in kwargs:
+            input_dict["fragment_charges"] = None
+        if "molecular_multiplicity" in kwargs and "fragment_multiplicities" not in kwargs:
+            input_dict["fragment_multiplicities"] = None
```

Comment:
probably if any _one_ of the four is present in kwargs, all need to be `None`d, since chgmult not independent.